frontend/catalyst/api_extensions/quantum_passes.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains public API functions that provide control for the
+user to input what MLIR compiler passes to run.
+
+Currently, each pass has its own user-facing decorator. In the future,
+a unified user interface for all the passes is necessary.
+
+Note that the decorators do not need to modify the qnode in
+any way. Its only purpose is to mark down the passes the user wants to
+run on each qnode, and then generate the corresponding
+transform.apply_apply_registered_pass in the lowered mlir.
+"""
+
+import pennylane as qml
+
+from catalyst.jax_primitives import apply_registered_pass_p, transform_named_sequence_p
+
+
+def inject_transform_named_sequence():
+    """
+    Inject a transform_named_sequence jax primitive.
+
+    This must be called when preprocessing the traced function in QJIT.capture(),
+    since to invoke -transform-interpreter, a transform_named_sequence primitive
+    must be in the jaxpr.
+    """
+
+    transform_named_sequence_p.bind()
+
+
+## API ##
+def cancel_inverses(fn=None):  # pylint: disable=line-too-long
+    """
+    Specify that a compiler pass for cancelling two neighbouring self-inverse
+    gates should be applied to the decorated QNode during qjit compilation.
+
+    This decorator is always applied to a qnode, and it cancels two neighbouring
+    self-inverse gates in the compiled mlir.
+
+
+    .. note::
+
+        Currently, only Hadamard gates are canceled.
+
+    .. note::
+
+        Unlike PennyLane :doc:`circuit transformations <introduction/compiling_circuits>`,
+        the QNode itself will not changed or transformed.
+
+        In other words, circuit inspection tools such as
+        :func:`~.draw` will still
+        display the neighbouring self-inverse gates. However, Catalyst never
+        executes the PennyLane code directly; instead, Catalyst captures the
+        workflow from Python and lowers it into MLIR, performing compiler
+        optimizations at the MLIR level.
+        To inspect the compiled MLIR from Catalyst, use
+        :func:`catalyst.debug.compiler_functions.print_compilation_stage`,
+        where ``stage="QuantumCompilationPass"``, and with ``keep_intermediate=True``
+        in the ``qjit`` decorator.
```

Comment:
```suggestion

        To inspect the compiled MLIR from Catalyst, use
        :func:`~..print_compilation_stage`,
        where ``stage="QuantumCompilationPass"``, and with ``keep_intermediate=True``
        in the ``qjit`` decorator.
```

Note that `~.` means two things:

- `~`: Sphinx will only display the function name in the link, not the full path
- `.`: Sphinx will search for the first matching function and cross reference it. Makes it more maintainable, since if someone moves the function it won't break the docstring link :)

Reply:
done (and good to know!)